### ISSUE TITLE
Error building with Cerbero

### ISSUE THREAD

**Issue by jwvdiermen** [EricssonResearch/openwebrtc#255]
Hi There,

First of all, thanks for the hard work, I really love what you have done with this library.

Sadly, I'm getting stuck compiling OpenWebRTC, which I need for an iOS app (I simply need the WebRTC API, I've not found any way to expose this on iOS via an easy to use Cordova plugin or anything. Isn't there a Cordova plugin that adds OpenWebRTC?). I followed all steps as described in here: https://github.com/EricssonResearch/openwebrtc/wiki/Building-OpenWebRTC-with-Cerbero But I get the error below after running for about an hour.

The command that fails is: `./cerbero-uninstalled -c config/osx-x86-64.cbc package openwebrtc`

I'm running this on a MacBook Pro with OS X 10.10.2. I don't have any experience with OS X, I mainly develop on Windows 7 and work with Linux.

The last bit of the console output:

```
[(39/41) openwebrtc -> fetch ]
[(39/41) openwebrtc -> extract ]
HEAD is now at 6db653c ./Makefile.am: build bindings last
Switched to branch 'cerbero_build'
HEAD is now at 6db653c ./Makefile.am: build bindings last
Cloning into '.'...
done.
Applying: Owr: Use videoconvert instead of ercolorspace on Android
Applying: Fix build on Android and iOS when not using build.sh
/Users/2gdeveloper/cerbero/sources/darwin_x86_64/openwebrtc-0.1/.git/rebase-apply/patch:113: trailing whitespace.
  dnl The GNU C++ STL is needed for JNI. This pkg-config file is
warning: 1 line adds whitespace errors.
Applying: build: Add a configure option to disable tests
error: patch failed: Makefile.am:1
error: Makefile.am: patch does not apply
Patch failed at 0001 build: Add a configure option to disable tests
The copy of the patch that failed is found in:
   /Users/2gdeveloper/cerbero/sources/darwin_x86_64/openwebrtc-0.1/.git/rebase-apply/patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
Running command 'git reset --hard origin/master'
Running command 'git branch cerbero_build'
Running command 'git checkout cerbero_build'
Running command 'git reset --hard origin/master'
Running command 'git clone /Users/2gdeveloper/cerbero/sources/local/openwebrtc -s -b cerbero_build .'
Running command 'git am --ignore-whitespace /Users/2gdeveloper/cerbero/recipes/openwebrtc/0001-Owr-Use-videoconvert-instead-of-ercolorspace-on-Andr.patch'
Running command 'git am --ignore-whitespace /Users/2gdeveloper/cerbero/recipes/openwebrtc/0002-Fix-build-on-Android-and-iOS-when-not-using-build.sh.patch'
Running command 'git am --ignore-whitespace /Users/2gdeveloper/cerbero/recipes/openwebrtc/0003-build-Add-a-configure-option-to-disable-tests.patch'
 
Recipe 'openwebrtc' failed at the build step 'extract'
Select an action to proceed:
[0] Enter the shell
[1] Rebuild the recipe from scratch
[2] Rebuild starting from the failed step
[3] Skip recipe
[4] Abort
```

I'd really appreciate any help you can give me, I'm currently stuck on this issue and am not able to create an iOS app with WebRTC support. 

The end goal is a simple app that will be used to connect to our existing WebRTC platform, so we can't use any existing libraries like PhoneRTC or OpenTok, that don't expose the WebRTC API, since we have our own signaling and API.

Thanks in advance!

**Comment by stefanalund** [comment]
Hi, we will soon (within a week or so) start to provide pre-build libraries for OpenWebRTC through a simple CocoaPod. 

If you want to understand how to build a hybrid app you can have a look at Bowser or openwebrtc-examples/ios/SimpleDemo

@superdump is looking in to the Cerbero build problems, I'll let him answer though.

**Comment by superdump** [comment]
We're actively working to iron out the last few issues with the cerbero builds on all platforms. As soon as we have done that and tested that all the example and test applications work as expected, we will move fully to that build system. We will also provide a release including binaries for download so that developers can get going more quickly.

We're then going to be working on automated builds, testing and bug fixing.

**Comment by jwvdiermen** [comment]
Ok, I'll wait a bit for you to move fully to the new build system and test again. Also very good news about the pre-build libraries, can't wait and really excited! Should I keep this issue open and report back when I'm able to build or just reopen if the issue persists after you've completely moved to Cerbero?

**Comment by stefanalund** [comment]
Let's keep it open. Would be great if you could retry at some point when Rob has fixed more (broken) things. Thanks!

**Comment by superdump** [comment]
https://github.com/EricssonResearch/openwebrtc/wiki/Building-OpenWebRTC

**Comment by nirbheek** [comment]
Is there an issue on this bug not already covered by other bugs? If not, we should close it. :)

**Comment by jwvdiermen** [comment]
I had some issues and errors during compiling, but just updating (as described here: https://github.com/EricssonResearch/openwebrtc/wiki/Building-OpenWebRTC#updating) and retrying fixed most issues. Build on Mac OS X for iOS finished, so did my build on Ubuntu 14.04 for Android. Building Android on Mac OS X failed, but I currently don't have the error handy since it wasn't a showstopper for me, since it worked on a Linux machine.

I also ran into this issue: https://github.com/EricssonResearch/cerbero/issues/1, but running `./cerbero-uninstalled -c config/cross-android-armv7.cbc buildone icu` fixed that issue.

As far as I'm concerned, this issue is now closed. Thanks! :)
